homeassistant/components/sonos/speaker.py

```diff
@@ -559,8 +555,22 @@ async def async_rebooted(self, soco: SoCo) -> None:
     #
     # Battery management
     #
-    async def async_update_battery_info(self, battery_dict: dict[str, Any]) -> None:
-        """Update battery info using the decoded SonosEvent."""
+    async def async_update_battery_info(self, more_info: str) -> None:
+        """Update battery info using a SonosEvent payload value."""
+        battery_dict = dict(x.split(":") for x in more_info.split(","))
```

Comment:
```suggestion
        battery_dict = {x.split(":") for x in more_info.split(",")}
```

Can you use the newer syntax here?